commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/input/CharCodeWithModifiers.java

```diff
@@ -64,4 +64,24 @@ public static int computeKeyDigest(int modifiers, int charCode) {
     public static int computeKeyDigest(SignalEvent event) {
         return computeKeyDigest(ModifierKeys.computeModifiers(event), KeyCodeMap.getKeyFromEvent(event));
     }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        CharCodeWithModifiers that = (CharCodeWithModifiers)o;
+
+        if (modifiers != that.modifiers) return false;
+        return charCode == that.charCode;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int hashCode() {
+        int result = modifiers;
+        result = 31 * result + charCode;
+        return result;
+    }
```

Comment:
as modifiers and charCode are final, I would avoid to compute hashCode everytime the method is invoked. hashcode result can be stored